Sources/System/FilePath/FilePath.swift

```diff
@@ -62,6 +62,18 @@ public struct FilePath {
 
 // @available(macOS 10.16, iOS 14.0, watchOS 7.0, tvOS 14.0, *)
 extension FilePath {
+  
+  /// Returns the current working directory of this process.
+  ///
+  /// - Warning: This value is global and care should be taken to make sure it does not unexpectedly change.
+  public static func getCurrentWorkingDirectoryForProcess() throws -> FilePath {
+    guard let cwd = system_getcwd(nil, 0) else {
```

Comment:
This will allocate twice -- wouldn't it be better if we supplied this with an appropriately sized FilePath buffer and only resize it if we guessed wrong?

If desired, we can also eliminate allocations altogether by adding a variant that takes an inout FilePath. (I don't think we need that.)

Reply:
This might be better but given all the discussion about how this shouldn't be used often, I preferred the simpler variant rather than adding an infrequently taken second codepath.

Reply:
`FilePath` used to have a with-style internal init that used max-path-length, but it turns out there are multiple notions of such a thing and it's probably too complicated as-is (and wastes space). I agree that it doesn't matter for now; in the future we'd like to do it with something like an init on `FilePath`, which can make the space/speed tradeoffs.